zebra-test/src/vectors.rs

```diff
@@ -165,4 +165,12 @@ lazy_static! {
     // https://github.com/zcash/zips/blob/master/zip-0243.rst#test-vector-3
     pub static ref ZIP243_3: Vec<u8> = <Vec<u8>>::from_hex("0400008085202f8901a8c685478265f4c14dada651969c45a65e1aeb8cd6791f2f5bb6a1d9952104d9010000006b483045022100a61e5d557568c2ddc1d9b03a7173c6ce7c996c4daecab007ac8f34bee01e6b9702204d38fdc0bcf2728a69fde78462a10fb45a9baa27873e6a5fc45fb5c76764202a01210365ffea3efa3908918a8b8627724af852fc9b86d7375b103ab0543cf418bcaa7ffeffffff02005a6202000000001976a9148132712c3ff19f3a151234616777420a6d7ef22688ac8b959800000000001976a9145453e4698f02a38abdaa521cd1ff2dee6fac187188ac29b0040048b004000000000000000000000000")
         .expect("Transaction bytes are in valid hex representation");
+
+    // V5 transaction test vectors
+    //
+    // This vectors are generated by our prop test system, they can be replaced with real mainnet and testnet
+    // blocks after Nu5 activates.
+    //
+    // https://github.com/oxarbitrage/documentation/blob/master/zebra/v5_test_vectors.md#generated_v5_1
+    pub static ref GENERATED_V5_1: Vec<u8> = <Vec<u8>>::from_hex(include_str!("vectors/block-generated-v5-001.txt").trim()).expect("Block bytes are in valid hex representation");
```

Comment:
How did you generate this test vector?

I think you might have found a bug in Zebra. (If that's the case, we'll want to fix the bug, then check that this test vector fails.)

You said:
> Current test vector haves height 1_046_401, we want to have this as 1. In order to do it we found out that 1_046_401 serialized and in hex string is 81F70F.
> So we replaced this with 010000 which is the hex serialized representation of 1u32.

https://github.com/oxarbitrage/documentation/blob/master/zebra/v5_test_vectors.md#hex

Those are little-endian integer bytes, but the block height is encoded in bitcoin script:
https://github.com/ZcashFoundation/zebra/blob/main/zebra-chain/src/transparent/serialize.rs#L44

Did you modify the height by changing `Coinbase.height` in code?
https://github.com/ZcashFoundation/zebra/blob/main/zebra-chain/src/transparent.rs#L84

Because you got a block that Zebra deserializes with a coinbase height of 1. If you can do it using little-endian integer bytes, then that's probably a bug in Zebra which we need to fix. (Or there's a bug in the test?)

Reply:
You are correct, we need to investigate further.
All i did here was obtaining a random prop test generated block. The hex string of it resulted in the one at https://github.com/oxarbitrage/documentation/blob/master/zebra/v5_test_vectors.md#hex

Then i searched for `81F70F` (you are correct this is little endian) and replaced it with the one that represent the number `1`. This worked so it seems like a bug.

Reply:
Can you print the coinbase input from the generated test vector?

It should be at `block.header.transactions[0].inputs()[0]` (or somewhere similar).

Here's the unmodified structure:
```json
Block {
    ...
    transactions: [
        V4 {
            inputs: [
                Coinbase {
                    height: Height(
                        1046401,
                    ),
                    data: CoinbaseData(
                        "g\\xe4\\xfaKUA_\\xa7\\x1a\\xa5\\xa3BaI*\\xbc\\xc6\\xfd3\\xe0x\\\\\\xca\\\"F\\x00",
                    ),
                    sequence: 1927961773,
                },
            ],
            ...
        },
    ],
}
```

Reply:
I removed the test vector from this PR as we are not going to use it. https://github.com/ZcashFoundation/zebra/pull/2185/commits/560f486c3d0114603b5781cccc06fc180b735b75

I will open a new draft PR for this and move the discussion there. Will post the new issue number here after this is done.

Reply:
https://github.com/ZcashFoundation/zebra/pull/2225